package.json

```diff
@@ -24,6 +24,7 @@
     "@angular/platform-browser": "4.1.3",
     "@angular/platform-browser-dynamic": "4.1.3",
     "@angular/router": "4.1.3",
+    "@angular/service-worker": "^1.0.0-beta.16",
```

Comment:
regarding service workers please read at https://developer.mozilla.org/en/docs/Web/API/Service_Worker_API or https://coryrylan.com/blog/fast-offline-angular-apps-with-service-workers
I will also write a blog post regarding that in susper.